### DISCUSSION TITLE
bg-[image-set()] can't be used?

### DISCUSSION THREAD

**Discussion by fishcy** [tailwindlabs/tailwindcss#11538] [Help]
for exmaple: bg-[image-set(url(imgUrl) 1x,url(imgUrl@2x) 2x)]

**Comment by wongjn** [comment]
`image-set()` *can* be used. For arbitrary variants, ensure you escape spaces in values by replacing them with an underscore:

```
bg-[image-set(url(imgUrl)_1x,url(imgUrl@2x)_2x)]
```

**Reply by fishcy** [reply]
ok, thanks

**Comment by roma-marshall** [comment]
Here a simpler way, try this: `bg-[url('/img/hero-pattern.svg')]`

source: https://tailwindcss.com/docs/background-image#customizing-your-theme

**Reply by fishcy** [reply]
ok, thanks

**Comment by MuhammadMujtabaa** [comment]
you may set the image URL in style prop instead of class
```
<div  style={{'--image-url': `url(${fetchedUrl})`}} >
</div>
```

**Reply by fishcy** [reply]
ok, thanks
